File: dotnet-adapters-shortest-paths-core/main/java/com/programmerare/shortestpaths/core/impl/GraphImpl.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.core.impl;

import java.util.List;

import com.programmerare.shortestpaths.core.api.Edge;
import com.programmerare.shortestpaths.core.api.Graph;
import com.programmerare.shortestpaths.core.api.Vertex;
import com.programmerare.shortestpaths.core.api.Weight;
import com.programmerare.shortestpaths.core.impl.generics.GraphGenericsImpl;
import com.programmerare.shortestpaths.core.validation.GraphEdgesValidationDesired;

public final class GraphImpl extends GraphGenericsImpl<Edge, Vertex , Weight> implements Graph {

	private GraphImpl(
		final List<Edge> edges,
		final GraphEdgesValidationDesired graphEdgesValidationDesired
	) {
		super(edges, graphEdgesValidationDesired);
	}

	/**
	 * Creates a graph instance, but will validate the edges and throw an exception if validation fails.
	 * If validation is not desired, then use the overloaded method. 
	 * @param edges list of all the edges for the graph
	 * @return an instance implementing the Graph interface
	 */	
	public static Graph createGraph(
		final List<Edge> edges
	) {
		return createGraph(edges, GraphEdgesValidationDesired.YES);
	}
	
	/**
	 * @param edges list of all the edges for the graph
	 * @param graphEdgesValidationDesired enum specifying whether or not validation is desired
	 * @return an instance implementing the Graph interface
	 */
	public static Graph createGraph(
		final List<Edge> edges,
		final GraphEdgesValidationDesired graphEdgesValidationDesired
	) {
		return new GraphImpl(edges, graphEdgesValidationDesired);
	}	
}

File: Programmerare.ShortestPaths.Core/src/com.programmerare.shortestpaths/core/impl/GraphImpl.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code in this "core" project is licensed with MIT.
* Other projects within this Visual Studio solution may be released with other licenses e.g. Apache.
* Please find more information in the files "License.txt" and "NOTICE.txt" 
* in the project root directory and/or in the solution root directory.
* It should also be possible to find more license information at this URL:
* https://github.com/TomasJohansson/adapters-shortest-paths-dotnet/
*/
using Programmerare.ShortestPaths.Core.Api;
using Programmerare.ShortestPaths.Core.Impl.Generics;
using Programmerare.ShortestPaths.Core.Validation;
using System.Collections.Generic;

namespace Programmerare.ShortestPaths.Core.Impl
{
    public sealed class GraphImpl : GraphGenericsImpl<Edge, Vertex , Weight> , Graph {

	    private GraphImpl(
		    IList<Edge> edges,
		    GraphEdgesValidationDesired graphEdgesValidationDesired
	    ) : base(edges, graphEdgesValidationDesired) {
	    }

        /// <summary>
        /// Creates a graph instance, but will validate the edges and throw an exception if validation fails.
        /// If validation is not desired, then use the overloaded method.
        /// </summary>
        /// <param name="edges">list of all the edges for the graph</param>
        /// <returns>an instance implementing the Graph interface</returns>
	    public static Graph CreateGraph(
		    IList<Edge> edges
	    ) {
		    return CreateGraph(edges, GraphEdgesValidationDesired.YES);
	    }
	
        /// <param name="edges">list of all the edges for the graph</param>
        /// <param name="graphEdgesValidationDesired">enum specifying whether or not validation is desired</param>
        /// <returns>an instance implementing the Graph interface</returns>
	    public static Graph CreateGraph(
		    IList<Edge> edges,
		    GraphEdgesValidationDesired graphEdgesValidationDesired
	    ) {
		    return new GraphImpl(edges, graphEdgesValidationDesired);
	    }	
    }
}